Classes/DataProcessing/MenuProcessor.php

```diff
@@ -53,48 +55,72 @@
  */
 class MenuProcessor extends \TYPO3\CMS\Frontend\DataProcessing\MenuProcessor
 {
+    use DataProcessingTrait;
+
     /**
+     * Allowed configuration keys for menu generation, other keys
+     * will throw an exception to prevent configuration errors.
+     *
      * @var array
      */
-    protected $menuLevelConfig = [
-        'doNotLinkIt' => '1',
-        'wrapItemAndSub' => '{|}, |*| {|}, |*| {|}',
-        'stdWrap.' => [
-            'cObject' => 'COA',
-            'cObject.' => [
-                '20' => 'TEXT',
-                '20.' => [
-                    'field' => 'nav_title // title',
-                    'trim' => '1',
-                    'wrap' => '"title":|',
-                    'preUserFunc' => 'TYPO3\CMS\Frontend\DataProcessing\MenuProcessor->jsonEncodeUserFunc'
-                ],
-                '21' => 'TEXT',
-                '21.' => [
-                    'value' => self::LINK_PLACEHOLDER,
-                    'wrap' => ',"link":|',
-                ],
-                '22' => 'TEXT',
-                '22.' => [
-                    'value' => self::TARGET_PLACEHOLDER,
-                    'wrap' => ',"target":|',
-                ],
-                '30' => 'TEXT',
-                '30.' => [
-                    'value' => '0',
-                    'wrap' => ',"active":|'
-                ],
-                '40' => 'TEXT',
-                '40.' => [
-                    'value' => '0',
-                    'wrap' => ',"current":|'
-                ],
-                '50' => 'TEXT',
-                '50.' => [
-                    'value' => '0',
-                    'wrap' => ',"spacer":|'
-                ]
-            ]
-        ]
+    public $allowedConfigurationKeys = [
+        'cache_period',
+        'entryLevel',
+        'entryLevel.',
+        'special',
+        'special.',
+        'minItems',
+        'minItems.',
+        'maxItems',
+        'maxItems.',
+        'begin',
+        'begin.',
+        'alternativeSortingField',
+        'alternativeSortingField.',
+        'showAccessRestrictedPages',
+        'showAccessRestrictedPages.',
+        'excludeUidList',
+        'excludeUidList.',
+        'excludeDoktypes',
+        'includeNotInMenu',
+        'includeNotInMenu.',
+        'alwaysActivePIDlist',
+        'alwaysActivePIDlist.',
+        'protectLvar',
+        'addQueryString',
+        'addQueryString.',
+        'if',
+        'if.',
+        'levels',
+        'levels.',
+        'expandAll',
+        'expandAll.',
+        'includeSpacer',
+        'includeSpacer.',
+        'as',
+        'titleField',
+        'titleField.',
+        'dataProcessing',
+        'dataProcessing.',
+        'appendData',
```

Comment:
The `appendData` property may need to be removed from HMENU configuration, by adding it also to the `$removeConfigurationKeysForHmenu` array.

https://github.com/TYPO3/TYPO3.CMS/blob/10.4/typo3/sysext/frontend/Classes/DataProcessing/MenuProcessor.php#L130-L148
https://github.com/TYPO3/TYPO3.CMS/blob/10.4/typo3/sysext/frontend/Classes/DataProcessing/MenuProcessor.php#L297-L303

Or did I understand that wrong?

Reply:
@theLine fixed. thx good catch! :)